File: backend/src/main/java/pt/isel/rugby/model/Athlete.java
package pt.isel.rugby.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "athlete")
@Data
public class Athlete implements Serializable{

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String height;

    @Column
    private String weight;

    @Column
    private String athleteNumber;

    @Column
    private String comment;

    @OneToOne
    @JoinColumn(name = "profileId", referencedColumnName = "id")
    private Profile profile;

    @OneToMany(mappedBy = "athlete")
    @JsonIgnore()
    private List<AthletePractice> athletePractices;

    @ManyToMany
    @JsonIgnore
    private List<TrainingSchedule> trainingSchedules;

    @ManyToMany
    @JsonIgnore
    private List<Game> games;


    @OneToMany(mappedBy = "athlete", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<ActiveRoster> activeRosters = new ArrayList();

    @Column
    private String positions;

    @OneToMany(mappedBy = "athlete")
    @JsonIgnore
    private List<AthleteGameStats> athleteGameStats;
    
}

File: frontend2/Rugby/src/app/classes/athlete.ts
import {Profile} from './profile';
import {Practice} from './practice';
import {TrainingSchedule} from './trainingschedule';
import {Game} from './game';
import {AthleteGameStats} from './associations/AthleteGameStats';

export class Athlete {
  constructor(
    public id?: number,
    public height?: string,
    public weight?: string,
    public athleteNumber?: string,
    public positions?: string,
    public comment?: string,
    public profile?: Profile,
    public practices?: Practice[],
    public trainingSchedule?: TrainingSchedule[],
    public games?: Game[],
    public athleteGameStats?: AthleteGameStats[]
  ) {
    this.id = id ? id : 0;
    this.height = height ? height : '';
    this.weight = weight ? weight : '';
    this.athleteNumber = athleteNumber ? athleteNumber : '';
    this.comment = comment ? comment : '';
    this.positions = positions ? positions : '';
    this.profile = profile ? profile : new Profile();
    this.practices = practices ? practices : [];
    this.trainingSchedule = trainingSchedule ? trainingSchedule : [];
    this.games = games ? games : [];
    this.athleteGameStats = athleteGameStats ? athleteGameStats : [];
  }
}


